proxy/envoy/discovery.go

```diff
@@ -356,6 +479,43 @@ func (ds *DiscoveryService) ListClusters(request *restful.Request, response *res
 	writeResponse(response, out)
 }
 
+// ListAllRoutes responds to RDS requests that are not limited by a key
+func (ds *DiscoveryService) ListAllRoutes(request *restful.Request, response *restful.Response) {
+
+	var allRoutes []routeConfigAndMetadata
+
+	endpoints := allServiceNodes(ds)
+
+	// This sort is not needed, but discovery_test excepts consistent output and sorting achieves it
+	sort.Strings(endpoints)
+
+	for _, ip := range endpoints {
+
+		instances := ds.services.HostInstances(map[string]bool{ip: true})
+		services := ds.services.Services()
+		httpRouteConfigs := buildOutboundHTTPRoutes(instances, services, &ProxyContext{
+			Discovery:  ds.services,
+			Config:     ds.config,
+			MeshConfig: ds.mesh,
+			IPAddress:  ip,
+		})
+
+		for port, httpRouteConfig := range httpRouteConfigs {
+
```

Comment:
(nit): extra newline here and elsewhere at the beginning of function and conditional blocks is not necessary.